driver-core/src/main/com/mongodb/internal/async/function/RetryState.java

```diff
@@ -204,10 +203,11 @@ private void doAdvanceOrThrow(final Throwable attemptException,
         if (isLastAttempt() || attemptException instanceof MongoOperationTimeoutException) {
```

Comment:
As we discussed on the call, only the following comment should be retained above this code block:

'It's important to check if the exception is an instance of MongoOperationTimeoutException to detect a timeout.'